server/controllers/users/changePassword.js

```diff
@@ -12,9 +12,10 @@ exports.changePassword = (req, res, next) => {
   }).then(([oldPasswordMatches, newPasswordMatches]) => {
     if (!oldPasswordMatches) throw new Error('You entered a wrong password');
     else if (newPasswordMatches) throw new Error('Your new password and old password are the same');
-    hash(newPassword, 10);
-    return changePassword(userId, newPassword);
-  }).then(() => res.send('Password is changed'))
+    return hash(newPassword, 10);
+  })
+    .then((hashedPassword) => changePassword(userId, hashedPassword))
+    .then(() => res.send('Password is changed'))
     .catch((err) => {
```

Comment:
https://github.com/GSG-G7/lang-mate/blob/8689ff7f37ae9c9c26a208b320cf2fbfd7b9ebf3/server/controllers/users/changePassword.js#L19
this should compare the message with the error messages we threw at lines 13 and 14
https://github.com/GSG-G7/lang-mate/blob/8689ff7f37ae9c9c26a208b320cf2fbfd7b9ebf3/server/controllers/users/changePassword.js#L13-L14

Reply:
```
if (err.message) next({ code: 400, msg: err.message });
else next(err);
```
This code for checking the error message

Reply:
all errors have a message tho, its what gets printed in the terminal when it happens along with the callstack tree